### PR TITLE
Graph state manager

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goals
* Factory to instantiate singleton for `GraphStateManager`
* Expose public functions to `import`, `act` and `export` 
* Add some basic `tests`

part of #22

### CODE REVIEW

**Comment by JoeCap08055** [comment]
Some comments:

* Since we're using NestJS, and I believe we want a singleton instance of GraphStateManager in the app, we should probably:
  * Decorate the class declaration with `@Injectable()`
  * Create a factory method to create the instance with the appropriate config
  * Use the factory method to inject into the app
  * Because of this, probably best to structure as its own sub-module 'graph-manager' (see `processor` sub-module for example)
  
  
* Since we want to batch chain updates, we probably want an easy way to simply iterate GraphStates and export all of the updates we've saved up.

File: src/graph/graph-state-manager.ts
Comment (saraswatpuneet) [inline src/graph/graph-state-manager.ts:94]: do we apply actions to all the states for that user?

File: src/graph/graph-state.module.ts
Comment (JoeCap08055) [inline src/graph/graph-state.module.ts:13]: Would be good to take this from the environment somehow

Comment (saraswatpuneet) [reply src/graph/graph-state.module.ts:13]: Yep will move it to configs

File: src/graph/graph-state.module.ts
Comment (saraswatpuneet) [inline src/graph/graph-state.module.ts:33]: @JoeCap08055  does this look right? using factory to inject `ConfigService`, not sure how to test this

Comment (JoeCap08055) [reply src/graph/graph-state.module.ts:10]: Hmmm... _looks_ right. You can inject the module into a jest test suite for testing. See `config.service.spec.ts` for an example.

Comment (saraswatpuneet) [reply src/graph/graph-state.module.ts:10]: Yep i did , check `graph-state-manager.spec.ts`

File: src/graph/graph-state-manager.spec.ts
Comment (JoeCap08055) [inline src/graph/graph-state-manager.spec.ts:50]: For efficiency, this can probably be `beforeAll`--I think it likely only needs to be done once, no? If we need to reset state after each test you can do an `afterEach` hook that clears the existing GraphStates without having to re-setup the entire app module under test.

Comment (saraswatpuneet) [reply src/graph/graph-state-manager.spec.ts]: Sure! just did

Comment (saraswatpuneet) [reply src/graph/graph-state-manager.spec.ts]: Since a finalizer is implemented for `GraphStateManager`, we would not need to free state explicitly

File: .env.docker.dev
Comment (johnmela) [inline .env.docker.dev:7]: Should we leave this to help with docker needed hosts in dev for `REDIS` and `FREQUENCY_URL`?

File: src/processor/reconnection-graph.service.ts
Comment (johnmela) [inline src/processor/reconnection-graph.service.ts:52]: I left these log statements in place until the connections could be pulled into graph SDK

File: src/graph/graph-state-manager.spec.ts
Comment (JoeCap08055) [reply src/graph/graph-state-manager.spec.ts]: I thought the finalizer didn't work?

File: src/graph/graph-state-manager.ts
Comment (JoeCap08055) [inline src/graph/graph-state-manager.ts:44]: This will be gone from GraphSDK once my latest PR gets merged

File: src/graph/graph-state-manager.ts
Comment (JoeCap08055) [inline src/graph/graph-state-manager.ts:11]: All references to capacity should be removed (see LIbertyDSNP/graph-sdk/#138)

**Comment by JoeCap08055** [comment]
This is marked as closing #22 , but I think there's more work to be done... this PR creates a class for working with the graph, but to close #22 we need to actually call the GraphStateManager from the GraphService and actually import/update/export/verify the graph.

File: src/graph/graph-state-manager.spec.ts
Comment (saraswatpuneet) [reply src/graph/graph-state-manager.spec.ts]: its at the module level, when `GraphStateManager` is garbage collected it calls the free on the state from sdk

Comment (saraswatpuneet) [reply src/graph/graph-state-manager.spec.ts]: the issue in node bridge is Graph class has not object to point to, since state is managed at rust side and all we have is a handle number, but in this case it works as when GraphStateManager is garbage collected it triggers the free state on finality

Comment (JoeCap08055) [reply src/graph/graph-state-manager.spec.ts]: ah, ok

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**
